powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/service/AuthenticationService.java

```diff
@@ -149,7 +149,21 @@ public CredentialAuthenticationResponse authenticateWithCredential(CredentialAut
         // Verify credential value
         final AuthenticationResult authenticationResult;
         if (credential.getStatus() == CredentialStatus.ACTIVE) {
-            authenticationResult = verifyCredential(request.getAuthenticationMode(), credential, credentialValue, request.getCredentialPositionsToVerify());
+            if (CredentialLocation.PROXY == credential.getSource()) {
```

Comment:
As of now, the method is already terribly long. May we extract to a method? More over, I still do not see any tests.